File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& arr, int target) {
        unordered_map<int,int>map;
        for(int i=0;i<arr.size();i++){
            int val=target-arr[i];
            if(map.find(val)!=map.end()){
                return {map[val]+1,i+1};
            }
            else{
                map[arr[i]]=i;
            }
        }
        return {};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        freq={}
        for i in range(len(nums)):
            freq[nums[i]]=i
        for i in range(len(nums)):
            diff=target-nums[i]
            if diff in freq:
                return [i+1,freq[diff]+1]
        return []

